addons/sale/data/mail_template_data.xml

```diff
@@ -214,7 +214,7 @@
                 <td>
                     <span style="font-weight:bold;">Shipping Method:</span>
                     <t t-out="object.carrier_id.name or ''"></t>
-                    <t t-if="object.carrier_id.fixed_price == 0.0">
+                    <t t-if="object.carrier_id.fixed_price == 0.0 or object.amount_delivery == 0.0">
```

Comment:
This fix isn't right:

- It will render as "Free" even if the fixed_price is 0 but the amount_delivery isn't.
- And it will render whatever fixed price is not matter what the actual amount_delivery is when both are above 0.

Reply:
Hi @chienandalu Thanks for the comments! I simply added another condition to avoid breaking something else. But when I relook at my old solution, it's indeed not clean enough. I will say a better fix will be:
```xml
                    <t t-if="object.amount_delivery == 0.0">
                        (Free)
                    </t>
                    <t t-else="">
                        (<t t-out="format_amount(object.amount_delivery, object.currency_id) or ''">$ 10.00</t>)
                    </t>
```

What do you think about it?

Reply:
Perfect! Thank you for the check! I will make a patch for it :smile: